app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/gainorlossafterreliefs/GlarCalculatorEligibilityUtil.scala

```diff
@@ -0,0 +1,151 @@
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.DisposalDetailsAnswers.CompleteDisposalDetailsAnswers
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.IndividualUserType.Self
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.ReliefDetailsAnswers.CompleteReliefDetailsAnswers
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.SingleDisposalTriageAnswers.CompleteSingleDisposalTriageAnswers
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.{CalculatedGlarBreakdown, DraftSingleDisposalReturn}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.repos.SessionStore
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.services.returns.ReturnsService
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.Logging
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class GlarCalculatorEligibilityUtil @Inject()(
+                                               val authenticatedAction: AuthenticatedAction,
+                                               val sessionDataAction: SessionDataAction,
+                                               val sessionStore: SessionStore,
+                                               val errorHandler: ErrorHandler,
+                                               returnsService: ReturnsService,
+                                               cc: MessagesControllerComponents,
+                                               servicesConfig: ServicesConfig
+                                             )(implicit ec: ExecutionContext)
+  extends FrontendController(cc)
+    with WithAuthAndSessionDataAction
+    with SessionUpdates
+    with Logging
+    with StartingToAmendToFillingOutReturnBehaviour {
+
+  val glarCalculatorFlag = servicesConfig.getBoolean("glar-calculator.enabled")
+
+  def withValuesForCalculatorWhereEligible(fillingOutReturn: FillingOutReturn)(
+    f: Option[CalculatedGlarBreakdown] => Future[Result]
+  )(implicit request: RequestWithSessionData[_]): Future[Result] = {
+
+    def isEligible(
+                    fillingOutReturn: FillingOutReturn,
+                    completeSingleDisposalTriageAnswers: CompleteSingleDisposalTriageAnswers,
+                    reliefDetailsAnswers: CompleteReliefDetailsAnswers
+                  )(f: Boolean => Future[Result]): Future[Result] =
+      if (!glarCalculatorFlag) {
+        f(false)
+      } else {
+        val eligibleAssetType              =
+          completeSingleDisposalTriageAnswers.assetType === Residential || completeSingleDisposalTriageAnswers.assetType === NonResidential
+
+        val noOtherReliefs                 = reliefDetailsAnswers.otherReliefs.isEmpty
+
+        val underPreviousReturnLimit       = fillingOutReturn.previousSentReturns
+          .getOrElse(PreviousReturnData(summaries = List.empty, previousYearToDate = None))
```

Comment:
this could have just been `fillingOutReturn.previousSentReturns.exists(_.summaries <= 10)`.